app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt

```diff
@@ -215,106 +210,103 @@ fun SearchBarScreen(
-                item {}
-                items(mSuggestionList, key = { it.keyword.hashCode() * 31 + it.canDelete.hashCode() }) {
-                    ListItem(
-                        headlineContent = { Text(text = it.keyword) },
-                        supportingContent = it.hint.ifNotNullThen { Text(text = it.hint!!) },
-                        leadingContent = it.canOpenDirectly.ifTrueThen {
-                            Icon(
-                                imageVector = Icons.AutoMirrored.Default.MenuBook,
-                                contentDescription = null,
-                            )
                         },
-                        trailingContent = it.canDelete.ifTrueThen {
-                            IconButton(onClick = { deleteKeyword(it.keyword) }, shapes = IconButtonDefaults.shapes()) {
-                                Icon(
-                                    imageVector = Icons.Default.Close,
-                                    contentDescription = null,
-                                )
+                        trailingIcon = {
+                            if (active) {
+                                AnimatedContent(targetState = searchFieldState.text.isNotEmpty()) { hasText ->
+                                    if (hasText) {
+                                        IconButton(onClick = { searchFieldState.clearText() }, shapes = IconButtonDefaults.shapes()) {
+                                            Icon(Icons.Default.Close, contentDescription = null)
+                                        }
+                                    } else {
+                                        IconButton(onClick = { navigate(ImageSearchScreenDestination) }, shapes = IconButtonDefaults.shapes()) {
+                                            Icon(Icons.Default.ImageSearch, contentDescription = null)
+                                        }
+                                    }
+                                }
+                            } else {
+                                Row {
+                                    trailingIcon()
+                                }
                             }
                         },
-                        colors = ListItemDefaults.colors(containerColor = Color.Transparent),
-                        modifier = Modifier.clickable { it.onClick() }.thenIf(animateItems) { animateItem() },
                     )
                 }
-            }
+                AppBarWithSearch(
+                    state = searchBarState,
+                    inputField = { inputField(false) },
+                    colors = SearchBarDefaults.appBarWithSearchColors(
+                        appBarContainerColor = Color.Transparent,
+                        scrolledAppBarContainerColor = Color.Transparent,
+                    ),
+                    scrollBehavior = scrollBehavior,
+                )
+                ExpandedFullScreenSearchBar(
```

Comment:
Adding quick search expands the search bar.
This [shit](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material3/material3/src/commonMain/kotlin/androidx/compose/material3/SearchBar.kt;l=538;drc=c4d396483225b0748703db85d49c9b44754f3b39) is so stupid I can't believe it passed code review.